src/components/TimeSelectInput/index.js

```diff
@@ -408,11 +464,20 @@ export default class TimeSelectInput extends Component {
         const minutesPlaceholder = this.prevMinutes || '--';
 
         return (
-            <article className={className} style={style} id={id}>
+            <StyledContainer className={className} style={style} id={id}>
+                <RenderIf isTrue={!!label}>
+                    <StyledLabel
+                        label={label}
+                        hideLabel={hideLabel}
+                        required={required}
+                        inputId="timeselectinput-label"
```

Comment:
we need to research here what is the best approach for accessibility since this is for associate the input with the label, we can't simply pass a fixed text here, also normally this value is generated and unique since it can't be used by another element in the DOM